### PR TITLE
refactor: use percent instead of fixed width for frequency count

### PR DESCRIPTION
**Pull request by johanlahti**
## Motivation

Improve the space management for frequency count while keeping frequency numbers intact (fixing customer bugs):
- Hard-code frequency width for the percent case, since `100.0%` is the longest string
- Use `flex-basic: max-content` and `min-width: max-content` where appropriate (so far, I have decided to use this in grid mode only, since we need to handle the space more efficiently)
- For absolute mode, increase the width so it can handle 8-10 digits
- Ensure a `min-width` exists for the value label (which also accounts for checkboxes) or it may disappear in some edge cases
- *New*: In single column mode and absolute count mode, use a consistent column width *when the value "touches" the frequency* (right-aligned for LTR and left-aligned for RTL)

*Note:* When Engine supports `qMaxGlyphCount` for frequency, we can switch to use this instead.

#### Demo with LTR: Notice how right-aligned is handled differently in grid mode (filling the space) which is good for using the space but can look slightly off when the frequency values varies a lot in size, thereby causing curved columns.

https://github.com/qlik-oss/nebula.js/assets/5780544/0e29e483-0da7-4e7a-b9e7-134f560fc885

## Requirements checklist

- [x] Api specification
  - [x] Ran `yarn spec`
    - [x] No changes **_OR_** API changes has been formally approved
- [x] Unit/Component test coverage
- [x] Correct PR title for the changes (fix, chore, feat)

When build and tests have passed:

- [x] Add code reviewers, for example @qlik-oss/nebula-core

### CODE REVIEW

**Comment by johanlahti** [comment]
#devbuild

File: apis/nucleus/src/components/listbox/utils/deduce-frequency-mode.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export default function deduceFrequencyMode(pages) {
-  const hasPercentSign = pages.some(({ qMatrix }) => qMatrix[0].some((item) => !!item.qFrequency?.match('%')));
+  const hasPercentSign = pages.some(({ qMatrix }) => qMatrix.some(([item]) => !!item.qFrequency?.match('%')));
```
Comment (T-Wizard) [inline apis/nucleus/src/components/listbox/utils/deduce-frequency-mode.js:2]: why this change?

File: apis/nucleus/src/components/listbox/components/ListBoxRowColumn/components/ListBoxRoot.jsx [hunk 2]
```diff
@@ -165,12 +174,12 @@ const RowColRoot = styled('div', {
...
     ...ellipsis,
-    // Frequency mode 'percent' will never need a text wider than this string: "100.0%"
-    flex: `0 0 ${getFreqFlexBasis({ sizes, frequencyMode, isGridMode })}`,
+    flex: `0 0 ${getFreqFlexBasis({ sizes, frequencyMode, isGridMode, freqHitsValue })}`,
     minWidth: 'auto',
+    // Frequency mode 'percent' will never need a text wider than this string: "100.0%"
     maxWidth: frequencyMode === 'P' ? `${sizes.freqMinWidth}px` : getMaxFreqWidth({ sizes, isGridMode }),
     textAlign: direction === 'rtl' ? 'left' : 'right',
-    paddingLeft: '2px',
+    paddingLeft: direction === 'rtl' ? '10px' : 0,
```
Comment (T-Wizard) [inline apis/nucleus/src/components/listbox/components/ListBoxRowColumn/components/ListBoxRoot.jsx]: why is there extra 10px padding only for rtl?

File: apis/nucleus/src/components/listbox/utils/deduce-frequency-mode.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export default function deduceFrequencyMode(pages) {
+  const hasPercentSign = pages.some(({ qMatrix }) => qMatrix.some(([item]) => !!item.qFrequency?.match('%')));
+  const OTHER = '-'; // treat all other as '-' (change later if we need to identify other modes)
+  return hasPercentSign ? 'P' : OTHER;
```
Comment (T-Wizard) [inline apis/nucleus/src/components/listbox/utils/deduce-frequency-mode.js:4]: this could be wrong if all values in pages has no frequency as they are excluded values.
is that OK? out could that cause problems with scrolling?

Comment (johanlahti) [reply apis/nucleus/src/components/listbox/utils/deduce-frequency-mode.js:4]: 🤷 Like I said before, the only workaround I can think of would be to get frequency mode from properties. Should we do that instead?
At the same time, it is more of a **visual** bug than a **functional** bug. I think it should be ok given that it is an edge case and will go away once we have frequency glyph count in the Engine response.

**Review by T-Wizard — APPROVED** [review]

File: apis/nucleus/src/components/listbox/utils/deduce-frequency-mode.js (hunk 3 above)
Comment (T-Wizard) [reply apis/nucleus/src/components/listbox/utils/deduce-frequency-mode.js:4]: It can also be seen when scrolling, but I agree that it is "more of a visual bug than a functional bug"
and we can probably live with it.
